checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallConsistencyAnalyzer.java

```diff
@@ -0,0 +1,1703 @@
+   * ownership of the result. Searches for the set of same resources in {@code obligations} and adds
+   * the new LocalVarWithTree to it if one exists. Otherwise creates a new set.
+   *
+   * @param obligations the currently-tracked obligations. This is always side-effected: an
+   *     obligation is either modified to include a new resource alias (the result of the invocation
+   *     being tracked) or a new resource alias set (i.e. obligation) is created and added.
+   * @param node the node whose result is to be tracked
+   */
+  private void trackInvocationResult(Set<ImmutableSet<LocalVarWithTree>> obligations, Node node) {
+    Tree tree = node.getTree();
+    // We need to track the result of the call iff there is a temporary variable for the call node
+    // (because we only create temporaries for expressions that actually have must-call values).
+    LocalVariableNode tmpVar = typeFactory.getTempVarForNode(node);
+    if (tmpVar == null) {
+      return;
+    }
+    LocalVarWithTree tmpVarWithTree = new LocalVarWithTree(new LocalVariable(tmpVar), tree);
+
+    // `mustCallAlias` is the MCA parameter if any exists, otherwise null.
+    Node mustCallAlias = getMustCallAliasParamVar(node);
+
+    // If mustCallAlias is null and call returns @This, set mustCallAlias to the receiver.
+    if (mustCallAlias == null
+        && node instanceof MethodInvocationNode
+        && typeFactory.returnsThis((MethodInvocationTree) tree)) {
+      mustCallAlias =
+          removeCastsAndGetTmpVarIfPresent(((MethodInvocationNode) node).getTarget().getReceiver());
+    }
+
+    if (mustCallAlias instanceof FieldAccessNode) {
+      // We do not track the call result if the MustCallAlias parameter is a field (handling of
+      // @Owning fields is a completely separate check, and we never need to track an alias of
+      // non-@Owning fields).
+    } else if (mustCallAlias instanceof LocalVariableNode) {
+      ImmutableSet<LocalVarWithTree> resourceAliasSetContainingMustCallAlias =
+          getResourceAliasSetForVar(obligations, (LocalVariableNode) mustCallAlias);
+      // If mustCallAlias is a local variable already being tracked, add tmpVarWithTree
+      // to the set containing mustCallAlias.
+      if (resourceAliasSetContainingMustCallAlias != null) {
+        ImmutableSet<LocalVarWithTree> newResourceAliasSet =
+            FluentIterable.from(resourceAliasSetContainingMustCallAlias)
+                .append(tmpVarWithTree)
+                .toSet();
+        obligations.remove(resourceAliasSetContainingMustCallAlias);
+        obligations.add(newResourceAliasSet);
+      }
+    } else {
+      // If mustCallAlias is neither a field nor a local already in the set of obligations,
+      // add it to a new set.
+      obligations.add(ImmutableSet.of(tmpVarWithTree));
+    }
+  }
+
+  /**
+   * Checks for cases where we do not need to track the result of a method call. An invocation
```

Comment:
Can you give an intuition for the concept of "need to track"?  Is it a synonym for "has an obligation"?  The documentation here does a good job of stating the specific conditions, but I would like to understand the concept.